modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -78,3 +82,61 @@ def _categories(self):  # noqa: GL08
     def _set_categories(self, categories):  # noqa: GL08
         self._lazy_categories = categories
         self._table = None
+
+
+def check_join_supported(join_type: str):
+    """
+    Check if join type is supported by HDK.
+
+    Parameters
+    ----------
+    join_type : str
+        Join type.
+
+    Returns
+    -------
+    None
+    """
+    if join_type not in ("inner", "left"):
+        raise NotImplementedError(
+            f"{join_type} join is not supported by the HDK engine"
+        )
+
+
+def arrow_to_pandas(at: pa.Table) -> pd.DataFrame:
+    """
+    Convert the specified arrow table to pandas.
+
+    Parameters
+    ----------
+    at : pyarrow.Table
+        The table to convert.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+
+    def mapper(at):
+        if is_dictionary(at) and isinstance(at.value_type, ArrowIntervalType):
+            # The default mapper fails with TypeError: unhashable type: 'dict'
+            return _CategoricalDtypeMapper
+        return None
+
+    return at.to_pandas(types_mapper=mapper)
+
+
+class _CategoricalDtypeMapper:  # noqa: GL08
+    @staticmethod
+    def __from_arrow__(arr):  # noqa: GL08
+        values = []
+        categories = OrderedDict()
+        chunks = arr.chunks if isinstance(arr, pa.ChunkedArray) else (arr,)
+        for chunk in chunks:
+            assert isinstance(chunk, pa.DictionaryArray)
+            cat = chunk.dictionary.to_pandas()
+            values.extend([cat[v.index.as_py()] if v.is_valid else None for v in chunk])
+            categories.update((c, None) for c in cat)
```

Comment:
I don't quite understand, why do we need to fill a dictionary with the keys all mapping to `None`?

Reply:
OrderedDict is used here as an ordered set.

Reply:
Let's add this as an in-code comment

Reply:
Done.